core/internal/flux_monitor_simulated_blockchain_test.go

```diff
@@ -0,0 +1,370 @@
+	minFee := app.Store.Config.MinimumContractPayment().ToInt().Int64()
+	require.Equal(t, fee, minFee, "fee paid by FluxAggregator (%d) must at "+
+		"least match MinimumContractPayment (%s). (Which is currently set in "+
+		"cltest.go.)", fee, minFee)
+
+	answer := int64(1) // Answer the nodes give on the first round
+
+	//- have one of the fake nodes start a round.
+	roundId := int64(1)
+	processedAnswer := answer * 100 /* job has multiply times 100 */
+	updateAnswer(t, answerParams{fa: &fa, roundId: roundId,
+		answer: processedAnswer, from: fa.neil, isNewRound: true,
+		completesAnswer: false})
+
+	// - successfully close the round through the submissions of the other nodes
+	// Response by malicious chainlink node, nallory
+	initialBalance := currentBalance(t, &fa).Int64()
+	reportPrice := answer
+	priceResponse := func() string {
+		return fmt.Sprintf(`{"data":{"result": %d}}`, reportPrice)
+	}
+	mockServer := cltest.NewHTTPMockServerWithAlterableResponse(t, priceResponse)
+	defer mockServer.Close()
+
+	// When event appears on submissionReceived, flux monitor job run is complete
+	submissionReceived := make(chan *faw.FluxAggregatorSubmissionReceived)
+	subscription, err := fa.aggregatorContract.WatchSubmissionReceived(nil,
+		submissionReceived, []*big.Int{}, []uint32{},
+		[]common.Address{fa.nallory.From})
+	require.NoError(t, err, "failed to subscribe to SubmissionReceived events")
+	defer subscription.Unsubscribe()
+
+	// Create FM Job, and wait for job run to start (the above UpdateAnswer call
+	// to FluxAggregator contract initiates a run.)
+	buffer := cltest.MustReadFile(t, "testdata/flux_monitor_job.json")
+	var job models.JobSpec
+	require.NoError(t, json.Unmarshal(buffer, &job))
+	initr := &job.Initiators[0]
+	initr.InitiatorParams.Feeds = cltest.JSONFromString(t, fmt.Sprintf(`["%s"]`,
+		mockServer.URL))
+	initr.InitiatorParams.PollingInterval =
+		models.Duration(timeout)
+	initr.InitiatorParams.Address = fa.aggregatorContractAddress
+	j := cltest.CreateJobSpecViaWeb(t, app, job)
+	jrs := cltest.WaitForRuns(t, j, app.Store, 1) // Submit answer from
+	reportedPrice := jrs[0].RunRequest.RequestParams.Get("result").String()
+	assert.Equal(t, reportedPrice, fmt.Sprintf("%d", reportPrice),
+		"failed to report correct price to contract")
+	var receiptBlock uint64
+	select { // block until FluxAggregator contract acknowledges chainlink message
+	case log := <-submissionReceived:
+		receiptBlock = log.Raw.BlockNumber
+	case <-time.After(timeout):
+		t.Fatalf("chainlink failed to submit answer to FluxAggregator contract")
+	}
+	checkUpdateAnswer(t, answerParams{fa: &fa, roundId: roundId,
+		answer: processedAnswer, from: fa.nallory, isNewRound: false,
+		completesAnswer: true}, initialBalance, receiptBlock)
+
+	//- have the malicious node start the next round.
+	nextRoundBalance := initialBalance - fee
+	// Triggers a new round, since price deviation exceeds threshold
+	reportPrice = answer + 1
```

Comment:
Is there any chance we could get a data race leading to intermittent test failures here?

Reply:
No, that's the role of the `select`s on either side of this instruction. They will wait for the node to respond to the reported change before allowing the test to continue (or time out and throw an error.)